src/utils.rs

```diff
@@ -48,4 +48,21 @@ pub unsafe fn as_slice<'a>(s: *const CHAR16) -> &'a [CHAR16] {
     }
 
     slice::from_raw_parts(s, len)
+}
+
+#[derive(Debug)]
+pub struct NullTerminatedAsciiStr<'a> {
+    buffer: &'a [u8]
+}
+
+impl<'a> NullTerminatedAsciiStr<'a> {
+    pub fn new(buffer: &[u8]) -> NullTerminatedAsciiStr {
+        assert!(buffer.len() >= 1);
+        assert!(buffer[buffer.len() - 1] == 0);
```

Comment:
I feel this function should be fallible. Therefore instead of using asserts we should have an if condition that returns error upon failure.

Reply:
Done